### PR TITLE
Add C# code to KinematicBody

### PR DESCRIPTION
**Pull request by skyace65**
Adds C# code to the KinematicBody page. This PR is WIP because we're not really sure what the C# equivalent to `preload` is and one code example doesn't work without that.

### CODE REVIEW

**Comment by karroffel** [comment]
@skyace65 `preload()` is a thing in GDScript that's deeply integrated into the parser of the language, there's no way to do that in other languages not made especially for/in Godot.

So instead of loading the resource at GDScript parse time, in C# you should use `ResourceLoader.Load()` instead. (I think, it's the same with GDNative, I assume that that's the same for C#)

**Comment by KellyThomas** [comment]
You should be able to use this to load the bullet scene:

    PackedScene bullet = (PackedScene) GD.Load("res://Bullet.tscn");

**Comment by skyace65** [comment]
@mysticfall The PR is finished, if you could review this that would be great!

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 1]
```diff
@@ -170,7 +177,42 @@ Attach a script to the KinematicBody2D and add the following code:
...
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class KBExample : KinematicBody2D
+    { 
+        int speed = 250;
+        Vector2 velocity = new Vector2();
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: Probably we should make `speed` a public property, and `velocity` a private field (with respective naming changes).

On a side note, I think declaring `velocity` as a member variable is sort of a bad practice. But I can see how it's duplicated from the GDScript counter part, so probably it needs to be dealt with a separate PR if we ever fix it.

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 2]
```diff
@@ -272,7 +360,39 @@ And the code for the Bullet:
...
+            {
+                velocity = velocity.Bounce(collsion.Normal);
+                if (collsion.Collider.HasMethod("Hit"))
+                {
+                    collsion.Collider.Hit();
+                }
+            }
+        }
+
+        public void onVisibilityNotifier2DScreenExited()
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: According to a discussion in #1365, it seems that we decided to follow PascalCase for signal callbacks. So, I believe we should treat it like any other methods regarding the naming convention, making it start with a upper case letter.

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 3]
```diff
@@ -337,7 +457,51 @@ Here's the code for the player body:
 
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class KBExample : KinematicBody2D
+    { 
+        [Export] int runSpeed = 100;
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: I guess you missed `public` here :)

**Comment by mysticfall** [comment]
@skyace65 Made a few comments regarding some inconsistencies I could find. Thanks for the heads up :)

**Comment by skyace65** [comment]
@mysticfall Updated the PR, can you check that I made the changes properly?

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 4]
```diff
@@ -170,7 +177,42 @@ Attach a script to the KinematicBody2D and add the following code:
...
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class KBExample : KinematicBody2D
+    { 
+        public int Speed = 250;
+        private Vector2 velocity = new Vector2();
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: Could you change it to `_velocity`? We recommend all private fields to be in camelCase with `_` prefix now.

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 5]
```diff
@@ -242,7 +284,53 @@ uses the mouse pointer. Here is the code for the Player, using ``move_and_slide(
 
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class KBExample : KinematicBody2D
+    { 
+        PackedScene bullet = (PackedScene)GD.Load("res://Bullet.tscn");
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: It's better to make access modifiers explicit, and in this case I guess we better use `private PackedScene _bullet` instead.

And the same goes for `public int Speed`, and `private Vector2 _velocity`.

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 6]
```diff
@@ -272,7 +360,39 @@ And the code for the Bullet:
 
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class Bullet : KinematicBody2D
+    { 
+        int speed = 750;
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: Same as above :)

File: tutorials/physics/using_kinematic_body_2d.rst [hunk 7]
```diff
@@ -337,7 +457,51 @@ Here's the code for the player body:
 
  .. code-tab:: csharp
 
-    Coming soon
+    using Godot;
+    using System;
+
+    public class KBExample : KinematicBody2D
+    { 
+        [Export] public int runSpeed = 100;
```
Comment (mysticfall) [inline tutorials/physics/using_kinematic_body_2d.rst]: public properties/fields need to be in PascalCase.

**Comment by mysticfall** [comment]
I only commented on the first occurrence of a similar issue, but it seems it has caused some confusion. I updated my comments to mark all such occurrences now. Sorry!

**Comment by skyace65** [comment]
Should be fixed now

**Comment by mysticfall** [comment]
It looks good to me. Thanks much, as always :)

**Comment by skyace65** [comment]
Thanks for your review :) @mhilbrunner feel free to merge whenever.

**Comment by cbscribe** [comment]
Thanks!

**Pull request merged**